discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -0,0 +1,378 @@
+            .order_by(desc(Play.slot))
+        ).first()
+        logger.error(f"index_solana_plays.py | get_latest_slot: {highest_slot_query}")
+        # Can be None prior to first write operations
+        if highest_slot_query is not None:
+            latest_slot = highest_slot_query.slot
+
+    # If no slots have yet been recorded, assume all are valid
+    if latest_slot is None:
+        latest_slot = 0
+
+    logger.info(
+        f"index_solana_plays.py | returning {latest_slot} for highest slot"
+    )
+    return latest_slot
+
+# Query a tx signature and confirm its existence
+def get_tx_in_db(session, tx_sig):
+    logger.info(f"index_solana_plays.py | checking db for {tx_sig}")
+    exists = False
+    tx_sig_db_count = (session.query(Play).filter(
+        Play.signature == tx_sig)).count()
+    exists = tx_sig_db_count > 0
+    logger.info(f"index_solana_plays.py | {tx_sig} exists={exists}")
+    return exists
+
+# pylint: disable=W0105
+'''
+Processing of plays through the Solana TrackListenCount program is handled differently
+than the original indexing layer
+
+Below we monitor the on chain 'programId' which is passed as a config - this will
+be deployed exactly once to Solana in conjunction with the AudiusEthRegistry
+
+Each transaction here is signed by a trusted ethereum address authorized within the audius
+protocol.
+
+Monitoring the address is performed by leveraging the `get_confirmed_signature_for_address2`
+function, which accepts 'limit' and 'before' parameters which are key to the logic below.
+This function returns tx signatures processed by the programId in confirmation order,
+with the most recently confirmed returned first.
+
+For example, if there are 1000 transactions and we request with a limit of 1000, the returned
+array will have the following format:
+[1000th, 999th, ..., 3rd, 2nd, 1]
+
+This indexing logic leverages the above to monitor the 'slots' for each transaction, based on
+the highest value currently stored in the discovery database and comparing that with the slots
+of the latest returned transactions until an intersection is explicitly found between the two.
+
+In the startup case, where there are no DB records present an intersection is considered 'found'
+when there are no records remaining - indicating that we have reached the end of the transaction
+history for this particular programId.
+
+For example, given the following state:
+
+latest_processed_slot = 200 <- latest slot stored in database
+
+last_tx_signature = None <- None by design, as we must start querying the most recent chain
+transactions
+
+transactions_history = get_confirmed_signature_for_address_2(before=None) =
+[
+    sig300, slot=230,
+    sig299, slot=230
+    sig298, slot=229
+    ...
+    sig250, slot=210
+]
+Since no intersection has been found, the above is loaded as a single 'batch' into an array of
+tx_signature batches that now has the following state:
+
+tx_batches = [batch1]
```

Comment:
for clarity can we name this batch like [batch_sig_300_to_250] or something like that? batch1 and batch2 are still a bit confusing as to the order